src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -44,27 +44,37 @@ Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
       RAY_CHECK_OK(actor_creator_->AsyncCreateActor(
           task_spec,
           [this, actor_id, task_id](Status status, const rpc::CreateActorReply &reply) {
-            if (status.ok()) {
-              RAY_LOG(DEBUG) << "Created actor, actor id = " << actor_id;
-              // Copy the actor's reply to the GCS for ref counting purposes.
+            if (status.ok() || status.IsCreationTaskError()) {
```

Comment:
We never had `status==IsCreationTaskError` previously. We always declare an actor creation task succeeds before event if `__init__` fails: 
- Since objects_valid is true, the PushTask always returns OK to GCS: https://github.com/ray-project/ray/blob/22a8ab6c061f04f83d73847ef97157e97d496351/src/ray/core_worker/transport/direct_actor_transport.cc#L188-L197
- Therefore, GCS actor scheduler will always see it succeds here: https://github.com/ray-project/ray/blob/1f3226ef866d1625c1d6d43ac553798dfbd941f5/src/ray/gcs/gcs_server/gcs_actor_scheduler.cc#L474-L486


We rely on the actor's worker to exit and pub-sub the actor death to GCS when actor creation task failed. 

This PR preserves this behavior by not retrying the actor creation task, but also didn't destroy the actor but wait for the coreworker runs the actor to exit eventually.